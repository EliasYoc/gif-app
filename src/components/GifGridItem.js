import React from "react";
import PropTypes from "prop-types";
export const GifGridItem = ({ img, title, id }) => {
  return (
    <article className="card animate__animated animate__fadeInLeft">
      <h3>{title}</h3>
      <img
        style={{ flexGrow: 1, objectFit: "contain" }}
        loading="lazy"
        src={img}
        alt={title}
      />
    </article>
  );
};

GifGridItem.propTypes = {
  img: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
