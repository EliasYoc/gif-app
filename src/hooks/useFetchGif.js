import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

const useFetchGif = (category) => {
  const [state, setState] = useState({ data: [], loading: true });
  useEffect(() => {
    getGifs(category).then((data) => setState({ data, loading: false }));
  }, [category]);
  return state;
};

export default useFetchGif;
