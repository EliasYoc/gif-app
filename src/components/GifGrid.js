import React from "react";
import useFetchGif from "../hooks/useFetchGif";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGif(category);
  return (
    <>
      <h3>{category}</h3>
      <section className="card-grid">
        {loading && <h5>Cargando...</h5>}
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </section>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
