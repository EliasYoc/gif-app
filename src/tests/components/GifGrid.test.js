import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import useFetchGif from "../../hooks/useFetchGif";
import "@testing-library/jest-dom";
jest.mock("../../hooks/useFetchGif");
describe("pruebas con <GifGrid/>", () => {
  const value = "goku";
  test("debe ejecutar el componente", () => {
    //simulando useFetch
    useFetchGif.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={value} />);

    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar items cuando se carga useFetch", () => {
    const gifs = [
      {
        id: "ABC",
        img: "https://localhost/cualquier.jpg",
        title: "lo que sea",
      },
      {
        id: "ABC",
        img: "https://localhost/cualquier.jpg",
        title: "lo que sea",
      },
    ];
    useFetchGif.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={value} />);
    expect(wrapper.find("h5").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
