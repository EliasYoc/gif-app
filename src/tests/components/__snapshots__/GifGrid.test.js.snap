// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con <GifGrid/> debe ejecutar el componente 1`] = `
<Fragment>
  <h3>
    goku
  </h3>
  <section
    className="card-grid"
  >
    <h5>
      Cargando...
    </h5>
  </section>
</Fragment>
`;
